### PR TITLE
Nix/Darwin NativePaths: Nix ignore -F flags and .tbl files

### PR DESCRIPTION
**Pull request by SkamDart**
Suggested fix for https://github.com/ziglang/zig/issues/19879

### CODE REVIEW

File: lib/std/zig/system/NativePaths.zig [hunk 1]
```diff
@@ -33,6 +33,10 @@ pub fn detect(arena: Allocator, native_target: std.Target) !NativePaths {
                     break;
                 };
                 try self.addFrameworkDir(framework_path);
+            } else if (mem.startsWith(u8, word, "-F")) {
+                // Ignore this argument.
+                // Nix provides the same framework path through -iframework
```
Comment (kubkon) [inline lib/std/zig/system/NativePaths.zig:38]: There is no harm in including duplicated paths, and it's strictly safer than outright ignoring `-F` arg. What if suddenly the sets are disjoint between everything passed via `-iframework` and `-F`?

File: lib/std/zig/system/NativePaths.zig [hunk 2]
```diff
@@ -67,6 +71,8 @@ pub fn detect(arena: Allocator, native_target: std.Target) !NativePaths {
                 try self.addRPath(lib_path);
             } else if (mem.startsWith(u8, word, "-l")) {
                 // Ignore this argument.
+            } else if (mem.endsWith(u8, word, ".tbd")) {
+                // Ignore this argument.
```
Comment (kubkon) [inline lib/std/zig/system/NativePaths.zig:75]: This should not be ignored, and actually handled in the linker. `tbd` is a textualised format for a dylib on macOS and skipping those may result in unexpected link failures.

**Review by kubkon — CHANGES_REQUESTED** [review]

File: lib/std/zig/system/NativePaths.zig (hunk 1 above)
Comment (SkamDart) [reply lib/std/zig/system/NativePaths.zig]: Valid point. I updated the PR to include that path instead of ignore it

**Comment by SkamDart** [comment]
@kubkon - Thanks for the review. I addressed your comments and rebased.

**Comment by andrewrk** [comment]
closing because it is still marked a draft with 30+ days of no updates.

**Pull request closed without merging**